java/src/com/ibm/streamsx/rest/primitives/JobGson.java

```diff
@@ -0,0 +1,39 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2017
+ */
+package com.ibm.streamsx.rest.primitives;
+
+import java.util.ArrayList;
+
+class JobGson {
```

Comment:
Gson is able to, as long as the class itself is simple enough.  If it gets too complex, we start needing to add in serializers & deserializers, or by setting up annotations on what is included/not included.  By de-coupling the actual json structure from what is used by the client applications, this is a cleaner implementation, and if the structure changes, it's impact is contained.